### PR TITLE
eslint fixes after eslint implementation

### PR DESCRIPTION
**Pull request by jacogr**
Depends on https://github.com/ethcore/eth-node-status-page/pull/129 (i.e. branched from there and fixes in this PR asked for there)

Builds on the above by fixing and removing the override warning rule from `eslintrc.json` for

- `no-return-assign`
- `react/jsx-no-bind`

Please check for validity & style as per your preferences.

### CODE REVIEW

File: client/components/Calls/index.js [hunk 1]
```diff
@@ -12,17 +12,29 @@ export default class Calls extends Component {
...
+    this.props.reset();
+  }
+
   render () {
     let {hoveredIdx} = this.state;
     return (
       <div
         className='calls-container'
-        onMouseLeave={() => this.setState({hoveredIdx: null})}
+        onMouseLeave={this._onMouseLeaveContainer}
```
Comment (tomusdrw) [inline client/components/Calls/index.js]: It doesn't need context binding here?

File: client/components/CallsToolbar/index.js [hunk 2]
```diff
@@ -27,6 +27,10 @@ export default class CallsToolbar extends Component {
       wrapStyle.right = 13;
     }
 
+    const setCall = () => ::this.setCall(call);
```
Comment (tomusdrw) [inline client/components/CallsToolbar/index.js]: `::` is superflous here (it is used to bind context, but here you are actually just invoking the method)

File: client/components/CallsToolbar/index.js [hunk 3]
```diff
@@ -27,6 +27,10 @@ export default class CallsToolbar extends Component {
...
+    const makeCall = () => ::this.makeCall(call);
```
Comment (tomusdrw) [inline client/components/CallsToolbar/index.js]: Same as above (I know it's copied from previous impl, but it would be better to fix it)

Comment (jacogr) [inline client/components/CallsToolbar/index.js]: Agreed. Didn't want to muck about too much, but it should be fixed - rather now than later.

File: client/components/Calls/index.js (hunk 1 above)
Comment (jacogr) [inline client/components/Calls/index.js]: Yeap, was too hasty, will fix.

**Comment by jacogr** [comment]
NightwatchCI fails for no apparent reason, on this and #129 

```
Error processing the server response: 
 Sauce Labs Authentication Error.
```

Probably since I'm an external contributor.

File: client/components/Calls/index.js [hunk 4]
```diff
@@ -12,17 +12,28 @@ export default class Calls extends Component {
...
       <div
         className='calls-container'
-        onMouseLeave={() => this.setState({hoveredIdx: null})}
+        onMouseLeave={onMouseLeaveContainer}
         {...this._test('container')}
       >
         {this.renderClear()}
         <h2 className={styles.header}>History</h2>
-        <div className={`${styles.history} row`} ref={el => this._callsHistory = el}>
+        <div className={`${styles.history} row`} ref={this._setCallsHistory}>
```
Comment (tomusdrw) [inline client/components/Calls/index.js]: Should be `ref={::this._setCallsHistory}` ? (to bind the context)

File: client/components/Calls/index.js [hunk 5]
```diff
@@ -12,17 +12,28 @@ export default class Calls extends Component {
...
   render () {
     let {hoveredIdx} = this.state;
+
+    const onMouseLeaveContainer = () => this.setState({hoveredIdx: null});
+
     return (
       <div
         className='calls-container'
-        onMouseLeave={() => this.setState({hoveredIdx: null})}
+        onMouseLeave={onMouseLeaveContainer}
```
Comment (tomusdrw) [inline client/components/Calls/index.js]: Could be a separate function and then `onMouseLeave={::this.onMouseLeaveContainer}`

File: client/components/Calls/index.js [hunk 6]
```diff
@@ -45,7 +56,7 @@ export default class Calls extends Component {
       <a
         {...this._test('remove')}
         title='Clear RPC calls history'
-        onClick={() => this.props.reset()}
+        onClick={this._clearHistory}
```
Comment (tomusdrw) [inline client/components/Calls/index.js]: Probably needs context binidng as well: `onClick={::this._clearHistory}`

File: client/components/Calls/index.js (hunk 5 above)
Comment (jacogr) [inline client/components/Calls/index.js]: I'm actually not 100% happy with the consistency here - in some cases using local functions (which makes 100% sense when dones as part of a callback, i.e. in a map), in other cases calling the class functions (and it is not shared anywhere)

It does lead to ::this issues (as you spotted correctly) and doesn't shout consistency across the codebase.

Comment (goldylucks) [inline client/components/Calls/index.js]: we don't have super strict rules about this.
for very short and simple functions like this hover one we usually do inline functions.

Comment (tomusdrw) [inline client/components/Calls/index.js]: I prefer to move everything as a class functions and use `bind` if arguments needs to be passed:
`onClick={this.onSomething.bind(this, idx)}`

Inline functions have tendency to get large and complicated (they can use whole environment). Having a lot of functions on class level is a clear indicator that some logic should be extracted into smaller component.

Comment (jacogr) [inline client/components/Calls/index.js]: My preference is for class functions as well, having everything like that would be cleaner and more maintainable. I will refactor if that is the consensus.

Comment (jacogr) [inline client/components/Calls/index.js]: *grumble* Took a quick look and the binding also creates unhappiness from the standard-react rules -

`react/jsx-no-bind JSX props should not use .bind()`

Will look further tonight. (Standard non-parameter functions are obviously ok)

Comment (tomusdrw) [inline client/components/Calls/index.js]: Yeah. I guess the suggestion is to split the item into separate component which is reasonable.

**Comment by tomusdrw** [comment]
Mind rebasing with master to see E2E passing?

**Pull request merged**